nixos/modules/services/networking/teleport.nix

```
@@ -0,0 +1,88 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.teleport;
+  pidFilePath = "/run/teleport.pid";
+  configJSON = pkgs.writeText "teleport.json" (builtins.toJSON cfg.settings);
+  cmdlineArgs = [ "--pid-file ${pidFilePath}" ]
+    ++ optional cfg.debug.enable "--debug"
+    ++ optional cfg.diag.enable "--diag-addr=${cfg.diag.addr}:${toString cfg.diag.port}"
+    ++ optional (cfg.settings != { }) "--config=${configJSON}"
+    ++ cfg.extraOpts;
+in
+{
+  options = {
+    services.teleport = with lib.types; {
+      enable = mkEnableOption "Teleport service.";
+
+      settings = mkOption {
+        type = attrs;
```

```suggestion
        type = (pkgs.formats.json {}).type;
```

I think `attr` will **eventually** be deprecated.